tensorrt_llm/executor/result.py

```diff
@@ -146,12 +152,104 @@ def logprobs_diff(self) -> TokenLogprobs | List[float]:
         return self.logprobs[self._last_logprobs_len:]
 
 
+def warmup_tensorrt_llm():
+    import tensorrt_llm
+    print("Warmup by importing tensorrt_llm with version",
+          tensorrt_llm.version.__version__)
+
+
+@ray.remote(max_concurrency=1000000, num_cpus=2)
+class RayAsyncQueue:
+    """Ray actor for async response handling."""
+
+    def __init__(self):
+        self.data = {}
+        self.event_map = {}
+        self.warmup_done = False
+
+    def register(self, key: int):
+        assert key not in self.event_map, f"Key {key} already registered"
+        self.event_map[key] = asyncio.Event()
+
+    def unregister(self, key: int):
+        if key in self.event_map:
+            del self.event_map[key]
+
+        if key in self.data:
+            del self.data[key]
+
+    def warmup(self):
+        if self.warmup_done:
+            return
+        warmup_tensorrt_llm()
+        self.warmup_done = True
+
+    def put_response(self, key: int, item: Any):
+        assert key in self.event_map, f"Key {key} not registered"
+        self.data[key] = item
+        self.event_map[key].set()
+
+    async def get_async(self, key: int):
+        assert key in self.event_map, f"Key {key} not registered"
+        await self.event_map[key].wait()
+        self.event_map[key].clear()
+        ret = self.data[key]
+        del self.data[key]
+        return ret
+
+
+SYNC_QUEUE_MAX_CONCURRENCY = 2
+
+
+@ray.remote(max_concurrency=SYNC_QUEUE_MAX_CONCURRENCY,
+            num_cpus=SYNC_QUEUE_MAX_CONCURRENCY)
+class RaySyncQueue:
+    """Ray actor for sync response handling."""
+
+    def __init__(self):
+        self.data = {}
+        self.event_map = {}
+        self.semaphore = threading.Semaphore(SYNC_QUEUE_MAX_CONCURRENCY - 1)
+        self.warmup_done = False
+
+    def register(self, key: int):
+        assert key not in self.event_map, f"Key {key} already registered"
+        self.event_map[key] = threading.Event()
+        self.event_map[key]
+
+    def unregister(self, key: int):
+        if key in self.event_map:
+            del self.event_map[key]
+
+        if key in self.data:
+            del self.data[key]
+
+    def warmup(self):
+        if self.warmup_done:
+            return
+        warmup_tensorrt_llm()
+        self.warmup_done = True
+
+    def put_response(self, key: int, item: Any):
+        self.data[key] = item
+        self.event_map[key].set()
+
+    def get(self, key: int):
+        with self.semaphore:
+            self.event_map[key].wait()
+            self.event_map[key].clear()
+            ret = self.data[key]
+            del self.data[key]
+            return ret
+
+
 class GenerationResultBase:
     ''' This holds the core logic of the GenerationResult class. '''
 
     def __init__(self,
                  id: int,
                  sampling_params: SamplingParams,
+                 queue: Optional[RayAsyncQueue] = None,
```

Comment:
Is this argument specific to Ray? To make it clearer, maybe we can either
- Renaming it to `ray_queue`
- Or, expose both `queue` and `aqueue` arguments for general purpose, and leave creating the aqueue and queue instances for different scenarios to the upper level.

Reply:
It's specific to Ray. Changing to `ray_queue`.